Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp

```diff
@@ -585,10 +589,50 @@ namespace AZ
+                //Only apply this optimization if you only have one view available.
+                if (renderPipeline && renderPipeline->GetViews(renderPipeline->GetMainViewTag()).size() == 1)
+                {
+                    RPI::ViewPtr cameraView = renderPipeline->GetDefaultView();
+                    const Matrix4x4& cameraWorldToClip = cameraView->GetWorldToClipMatrix();
+                    worklistData->m_cameraFrustum = Frustum::CreateFromMatrixColumnMajor(cameraWorldToClip);
+                    worklistData->m_applyCameraFrustumIntersectionTest = true;
+                }
             }
             
             auto nodeVisitorLambda = [worklistData, taskGraph, parentJob, &worklist](const AzFramework::IVisibilityScene::NodeData& nodeData) -> void
             {
+                // For shadow cascades that are greater than index 0 we can do another check to see if we can reject any Octree node that do not
+                // intersect with the camera frustum. We do this by checking for an overlap between the camera frustum and the Obb created
+                // from the node's AABB but rotated and extended towards light direction. 
+                if (worklistData->m_applyCameraFrustumIntersectionTest && worklistData->m_hasExcludeFrustum)
+                {
+                    // Build an Obb from the Octree node's aabb
+                    AZ::Obb extrudedBounds = AZ::Obb::CreateFromAabb(nodeData.m_bounds);
+
+                    // Rotate the Obb in the direction of the light
+                    AZ::Quaternion directionalLightRot = worklistData->m_view->GetCameraTransform().GetRotation();
+                    extrudedBounds.SetRotation(directionalLightRot);
+                    
+                    AZ::Vector3 halfLegth = 0.5f * nodeData.m_bounds.GetExtents();
+                    // After converting AABB to OBB we apply a rotation and this can incorrectly fail intersection test. If you have an OBB cube built from an octree node,
+                    // rotating it can cause it not encapsulate meshes it encapsulated beforehand. The type of shape we want here is essentially a capsule that starts from the
+                    // light and wraps the aabb of the octree node cube and extends towards light direction. This capsule's diameter needs to the size of the body diagonal
+                    // of the cube. Since using capsule shape will make intersection test expensive we simply expand the Obb to have each side be atleast the size of the body diagonal
+                    // which is sqrt(3) * side size. Hence we expand the Obb by 73%. 
+                    halfLegth += halfLegth * Vector3(0.36f, 0.36f, 0.36f);
+                    
+                    // Next we extrude the Obb in the direction of the light in order to ensure we capture meshes that are behind the camera but cast a shadow within it's frustum
+                    halfLegth.SetY(halfLegth.GetY() + r_shadowCascadeExtrutionAmount);
```

Comment:
if you have a directional light component on an entity, is "behind the camera" relative to the entity world space position?  and is the r_shadowCascadeExtrusionAmount in meters in the -Y direction relative to the (shadow) camera?  it's unclear to me when this parameter would need to be changed, but it seems if you had a directional light on an entity that was in the middle of your scene and a building more than 20 meters "behind" the directional light, it would no longer be drawn into the shadow cascade.

Reply:
- Correct, "Behind the Camera" is relative to the entity's world space position. 
- Correct, r_shadowCascadeExtrusionAmount is a value in meters in the forward direction which is +Y (O3de is Z-up in world space).
- Correct, consider this cvar as extra padding but it is not a perfect solution. We already expand the Obb in the y-axis by 36% so it will capture a lot of the objects in the near vicinity. I added support for this cvar in case people are still missing shadows for an object that was much further away and outside the frustum and want to extrude the Obb even further they can modify the cvar. We dont want to make this cvar too large as then you are loosing the perf gain by this optimizaiton.